packages/amplify_api/android/src/test/kotlin/com/amazonaws/amplify/amplify_api/AmplifyApiRestTest.kt

```diff
@@ -0,0 +1,429 @@
+            )
+            mockRestOperation
+        }.`when`(mockApi).get(
+                any(RestOptions::class.java),
+                any(),
+                any())
+
+        var arguments : Map<String, Any> = mapOf(
+                "restOptions" to mapOf(
+                    "path" to "/items"
+                ),
+                "cancelToken" to "someCode"
+        )
+
+        flutterPlugin.onMethodCall(
+                MethodCall("get", arguments),
+                mockResult
+        )
+
+        // Verify input RestOptions
+        val optionsCaptor = ArgumentCaptor.forClass(RestOptions::class.java)
+        verify(mockApi).get(
+                optionsCaptor.capture(),
+                any<Consumer<RestResponse>>(),
+                any<Consumer<ApiException>>()
+        )
+        assertEquals(
+                RestOptions.builder().addPath("/items").build(),
+                optionsCaptor.value
+        )
+
+        verify(mockResult).success(
+                mapOf(
+                        "data" to restResponse.data.rawBytes
+                )
+        )
+
+    }
+
+    @Test
+    fun test_post_returns_success(){
+
+        var body = toStoreData
+        var data = postSuccessData
+
+        var restResponse = RestResponse(200, data)
+
+        Mockito.doAnswer { invocation ->
+            (invocation.arguments[2] as Consumer<RestResponse>).accept(
+                    restResponse
+            )
+            mockRestOperation
+        }.`when`(mockApi).post(
+                any(), // NOTE that passing "apiName" causes 4 parameter Amplify.API function to be called !!!
+                any(RestOptions::class.java),
+                any(),
+                any())
+
+        var arguments : Map<String, Any> = mapOf(
+                "restOptions" to mapOf(
+                        "path" to "/items",
+                        "body" to body,
+                        "apiName" to "restapi",
+                        "headers" to mapOf(
+                                "headerA" to "headerValueA",
+                                "headerB" to "headerValueB"
+                        ),
+                        "queryParameters" to mapOf(
+                                "queryParameterA" to "queryValueA",
+                                "queryParameterB" to "queryValueB"
+                        )
+                ),
+                "cancelToken" to "someCode"
+        )
+
+        flutterPlugin.onMethodCall(
+                MethodCall("post", arguments),
+                mockResult
+        )
+
+        // Verify input RestOptions
+        val optionsCaptor = ArgumentCaptor.forClass(RestOptions::class.java)
+        verify(mockApi).post(
+                any(),
+                optionsCaptor.capture(),
+                any<Consumer<RestResponse>>(),
+                any<Consumer<ApiException>>()
+        )
+        // While the RestOptions are equal, a Kotlin bug registers the results as different
+        /*
```

Comment:
Can you add more context on the kotlin bug here so it's helpful for writing tests in other places.